modules/currency-l1/src/main/scala/org/tessellation/currency/l1/DataApplication.scala

```diff
@@ -54,6 +63,20 @@ object DataApplication {
     def ownRoundTrigger: Stream[F, OwnerConsensusInput] =
       Stream
         .awakeEvery(5.seconds)
+        .evalMap { _ =>
```

Comment:
Not actually. I don't know much about the semaphore, the previous version didn't use the semaphore in any places so I thought it might not be used. Do you think we should add? @marcinwadon

Reply:
In the `StateChannel` [class](https://github.com/Constellation-Labs/tessellation/blob/49bb9209ee289652d24de340efae46c04bb5f3e3/modules/dag-l1/src/main/scala/org/tessellation/dag/l1/StateChannel.scala) there are `Semaphore(1)` instances created in the `make` method that are then used in the outer function calling `canStartOwnConsensus`. As I understand, Semaphore(1) is equivalent to a Mutex, which is just an explicit lock on an IO resource. I'm also still learning the nuance but I believe the lock is needed here since we have multiple processes which need to be in an "open" status (therefore stable and not in the middle of a modification) before going about the consensus operation.

I think you may also want to import `org.tessellation.ext` to access the `.evalMapLocked` helper method to recreate the process from StateChannel

Reply:
I think it's because the currency accepts and stores transactions but we do not store data updates, we just send them to L0 for consensus acceptance. So L1 here is kind a proxy. I'm not sure about the function you've just added, hence the question. The currency consensus uses `evalMapLocked` with block acceptance semaphore (I bet it's a mutex).

Reply:
I think that too, Marcin